### PR TITLE
update user details

### PR DESCRIPTION
**Pull request by Rishab-innovative**

### CODE REVIEW

File: src/pages/EditDetailsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const updatedRegisterData = {
+      ...updatedData,
+      [event.target.id]: event.target.value,
+    };
+    setUpdatedData(updatedRegisterData);
+  };
+
+  useEffect(() => {
```
Comment (PiyushInnotech) [inline src/pages/EditDetailsPage.tsx:66]: remove these

File: src/pages/EditDetailsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      dispatch(
+        SaveEditedProfilePicture({
+          uid: LoggedInUserData.userDetails.uid,
+          DocId: LoggedInUserData.userDetails.DocId,
+          picture: updatedData.picture,
+        })
+      );
+    }
+  };
+  const handleSuccessSignUp = () => {
```
Comment (PiyushInnotech) [inline src/pages/EditDetailsPage.tsx:113]: change name

File: src/redux/EditDetailsSlice.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        mobileNumber: data.mobileNumber,
+      });
+    } catch (error) {
+      console.error("Error Posting user details:", error);
+      throw error;
+    }
+  }
+);
+
+export const SaveEditedProfilePicture = createAsyncThunk(
```
Comment (PiyushInnotech) [inline src/redux/EditDetailsSlice.tsx:39]: use same function fpr both

File: src/pages/EditDetailsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    navigate("/userProfile");
+    dispatch(resetSuccess());
+  };
+  return (
+    <>
+      {updateDetailStatus.status === "succeeded" ? (
+        <Modal open={openCloseModal} onClose={() => setOpenCloseModal(false)}>
```
Comment (PiyushInnotech) [inline src/pages/EditDetailsPage.tsx:120]: create separate component for successmodal with dynamic

File: src/App.tsx [hunk 5]
```diff
@@ -1,16 +1,86 @@
...
+  Routes,
+  Route,
+  Navigate,
+} from "react-router-dom";
 import "./App.css";
-import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 import LoginPage from "./pages/LoginPage";
 import SignupPage from "./pages/SignupPage";
 import UserProfilePage from "./pages/UserProfilePage";
+import EditDetailsPage from "./pages/EditDetailsPage";
```
Comment (meanstack9716) [inline src/App.tsx:12]: EditUserDetil page @Rishab-innovative

File: src/App.tsx [hunk 6]
```diff
@@ -1,16 +1,86 @@
...
+import Navbar from "./components/Navbar";
+import { onAuthStateChanged } from "firebase/auth";
+import { auth } from "./firebase";
+import { useDispatch } from "react-redux";
+import { AppDispatchType, RootState } from "./redux/Store";
+import { fetchUserDetails } from "./redux/NavBarSlice";
+import ProtectedRoute from "./ProtectedRoute";
+import { Box, CircularProgress } from "@mui/material";
+import { useSelector } from "react-redux";
```
Comment (meanstack9716) [inline src/App.tsx:21]: always import all library on the top, the all our components and services @Rishab-innovative

Comment (meanstack9716) [reply src/App.tsx:21]: this is for all pages @Rishab-innovative

File: src/App.tsx [hunk 7]
```diff
@@ -1,16 +1,86 @@
...
 
 function App() {
+  const dispatch = useDispatch<AppDispatchType>();
+  const SuccessSignUp = useSelector(
```
Comment (meanstack9716) [inline src/App.tsx:25]: variable name should start with lowerCase @Rishab-innovative

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

**Pull request merged**